docs/ternary-operator.md

```diff
@@ -0,0 +1,165 @@
+
+```BrighterScript
+a = user = invalid ? "not logged in" : "logged in"
+```
+
+transpiles to:
+
+```BrightScript
+a = bslib_ternary(user = invalid, "not logged in", "logged in")
+```
+
+```BrighterScript
+a = user = invalid ? defaultUser : user
+```
+
+transpiles to:
+
+```BrightScript
+a = bslib_ternary(user = invalid, defaultUser, user)
+```
+
+### Scope capturing
+In this situation, BrighterScript has detected that your ternary operation will have side-effects or could possibly result in a rendezvous. BrighterScript will create an immediately-invoked-function-expression to capture all of the referenced local variables. This is in order to only execute the consequent if the condition is true, and only execute the alternate if the condition is false.
+
+```BrighterScript
+  a = user = invalid ? "no name" : user.name
+```
+
+transpiles to:
+
+```BrightScript
+a = (function(__bsCondition, user)
+        if __bsCondition then
+            return "no name"
+        else
+            return user.name
+        end if
+    end function)(user = invalid, user)
+```
+
+```BrighterScript
+a = user = invalid ? getNoNameMessage(m.config) : user.name + m.accountType
+```
+
+transpiles to:
+
+```BrightScript
+a = (function(__bsCondition, getNoNameMessage, m, user)
+        if __bsCondition then
+            return getNoNameMessage(m.config)
+        else
+            return user.name + m.accountType
+        end if
+    end function)(user = invalid, getNoNameMessage, m, user)
+```
+
+### nested scope protection
+The scope protection works for multiple levels as well
+```BrighterScript
+m.count = 1
+m.increment = function ()
+    m.count++
+    return m.count
+end function
+result = (m.increment() = 1 ? m.increment() : -1) = -1 ? m.increment(): -1
+```
+
+transpiles to:
+```BrightScript
+m.count = 1
+m.increment = function()
+    m.count++
+    return m.count
+end function
+result = (function(__bsCondition, m)
+        if __bsCondition then
+            return m.increment()
+        else
+            return - 1
+        end if
+    end function)(((function(__bsCondition, m)
+            if __bsCondition then
+                return m.increment()
+            else
+                return - 1
+            end if
+        end function)(m.increment() = 1, m)) = - 1, m)
+```
+
+
+## Library functions
+
+The following library functions are called in the transpiled code:
+
+```
+function bslib_ternary(isTrue, trueValue, falseValue)
+    if isTrue then
+        return trueValue
+    else
+        return falseValue
+    end if
+end function
+```
+
+## No standalone statements
+The BrighterScript ternary operator differs from languages like C# and JavaScript in that you cannot use ternary expressions as standalone expressions. This is due to the fact that BrightScript uses `=` for both assignment and equality. As such, ternary expressions must always be part of an assignment or method call. Take a look at the following expression:
+```
+a = myValue ? "a" : "b"
```

Comment:
I think we want to say that in a simpler way like: 
```
' this is not a valid statement
condition ? doSomething() : orElse()
```
Due to Brightscript syntax where an assignment like `a = b` can not be differentiated from a condition `if a = b`.

Reply:
BTW wouldn't that be ok?
```
' expression hanging alone
(condition ? doSomething() : orElse())
```